source/common/statistic_impl.h

```diff
@@ -150,4 +151,118 @@ class HdrStatistic : public StatisticImpl {
   struct hdr_histogram* histogram_;
 };
 
-} // namespace Nighthawk
\ No newline at end of file
+/**
+ * CircllhistStatistic uses Circllhist under the hood to compute statistics.
+ * Circllhist is used in the implementation of Envoy Histograms, compared to HdrHistogram it trades
+ * precision for fast performance in merge and insertion. For more info, please see
+ * https://github.com/circonus-labs/libcircllhist
+ */
+class CircllhistStatistic : public StatisticImpl {
+public:
+  CircllhistStatistic() {
```

Comment:
Can we place the implementation of the constructor in the .cc file? It helps to keep the header files more readable and focused on the API.

Same comment applies to the destructor and the other methods and class.

Reply:
Done. PTAL